dotnet/src/Connectors/Connectors.MistralAI/Client/MistralClient.cs

```diff
@@ -0,0 +1,897 @@
+            }
+            if (toolCalls.Count > 0)
+            {
+                message.ToolCalls = [.. toolCalls.Values];
+            }
+            return [message];
+        }
+
+        if (content.Role == AuthorRole.Tool)
+        {
+            List<MistralChatMessage>? messages = null;
+            foreach (var item in content.Items)
+            {
+                if (item is not FunctionResultContent resultContent)
+                {
+                    continue;
+                }
+
+                messages ??= [];
+
+                var stringResult = ProcessFunctionResult(resultContent.Result ?? string.Empty, toolCallBehavior);
+                messages.Add(new MistralChatMessage(content.Role.ToString(), stringResult));
+            }
+            if (messages is not null)
+            {
+                return messages;
+            }
+
+            throw new NotSupportedException("No function result provided in the tool message.");
+        }
+
+        return [new MistralChatMessage(content.Role.ToString(), content.Content ?? string.Empty)];
+    }
+
+    private HttpRequestMessage CreatePost(object requestData, Uri endpoint, string apiKey, bool stream)
+    {
+        var httpRequestMessage = HttpRequest.CreatePostRequest(endpoint, requestData);
+        this.SetRequestHeaders(httpRequestMessage, apiKey, stream);
+
+        return httpRequestMessage;
+    }
+
+    private void SetRequestHeaders(HttpRequestMessage request, string apiKey, bool stream)
+    {
+        request.Headers.Add("User-Agent", HttpHeaderConstant.Values.UserAgent);
+        request.Headers.Add(HttpHeaderConstant.Names.SemanticKernelVersion, HttpHeaderConstant.Values.GetAssemblyVersion(this.GetType()));
+        request.Headers.Add("Accept", stream ? "text/event-stream" : "application/json");
+        request.Headers.Add("Authorization", $"Bearer {apiKey}");
+        request.Content!.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+    }
+
+    private async Task<T> SendRequestAsync<T>(HttpRequestMessage httpRequestMessage, CancellationToken cancellationToken)
+    {
+        using var response = await this._httpClient.SendWithSuccessCheckAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+
+        var body = await response.Content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false);
+
+        return DeserializeResponse<T>(body);
+    }
+
+    private async Task<HttpResponseMessage> SendStreamingRequestAsync(HttpRequestMessage httpRequestMessage, CancellationToken cancellationToken)
+    {
+        return await this._httpClient.SendWithSuccessCheckAsync(httpRequestMessage, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
+    }
+
+    private Uri GetEndpoint(MistralAIPromptExecutionSettings executionSettings, string path)
+    {
+        var endpoint = this._endpoint ?? new Uri($"https://api.mistral.ai/{executionSettings.ApiVersion}");
```

Comment:
Move to constant: "https://api.mistral.ai/"
```suggestion
        var endpoint = this._endpoint ?? new Uri($"{PublicMistralAPIBaseUri}{executionSettings.ApiVersion}");
```
```suggestion
        var endpoint = this._endpoint ?? new Uri($"https://api.mistral.ai/{executionSettings.ApiVersion}");
```